newrelic/hooks/mlmodel_sklearn.py

```diff
@@ -80,12 +80,52 @@ def _nr_wrapper_method(wrapped, instance, args, kwargs):
         if method in ("predict", "fit_predict"):
             training_step = getattr(instance, "_nr_wrapped_training_step", "Unknown")
             wrap_predict(transaction, _class, wrapped, instance, args, kwargs)
+            create_label_event(transaction, _class, instance, return_val)
             return PredictReturnTypeProxy(return_val, model_name=_class, training_step=training_step)
         return return_val
 
     wrap_function_wrapper(module, "%s.%s" % (_class, method), _nr_wrapper_method)
 
 
+def create_label_event(transaction, _class, instance, return_val):
+    inference_id = uuid.uuid4()
+    model_name = getattr(instance, "_nr_wrapped_name", _class)
+    model_version = getattr(instance, "_nr_wrapped_version", "0.0.0")
+    label_names = getattr(instance, "_nr_wrapped_label_names", None)
+    settings = transaction.settings if transaction.settings is not None else global_settings()
+
+    if return_val is not None:
+        for index, value in enumerate(return_val):
+            python_value_type = str(type(value))
+            value_type = str(categorize_data_type(python_value_type))
+            label_names_list = _get_label_names(label_names, return_val)
```

Comment:
We should move this call outside of the loop so it's not re-computing the names for each value.